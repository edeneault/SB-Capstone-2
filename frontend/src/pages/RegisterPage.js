import React, { useState, useEffect } from "react";
import { Link } from "react-router-dom";
import { Form, Button, Row, Col } from "react-bootstrap";
import { useDispatch, useSelector } from "react-redux";
import Loader from "../components/Loader";
import FormContainer from "../components/FormContainer";
import { register } from "../actions/userActions";

import { toast } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";

toast.configure();

const RegisterScreen = ({ location, history }) => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [name, setName] = useState("");
  const [confirmPassword, setConfirmPassword] = useState("");
  const [message, setMessage] = useState(null);
  const dispatch = useDispatch();

  const userRegister = useSelector((state) => state.userRegister);

  const { loading, error, userInfo } = userRegister;

  const redirect = location.search ? location.search.split("=")[1] : "/";

  useEffect(() => {
    if (userInfo) {
      history.push(redirect);
    }
  }, [history, userInfo, redirect]);

  const submitHandler = (e) => {
    e.preventDefault();
    if (password !== confirmPassword) {
      setMessage("Passwords do not match");
    } else {
      dispatch(register(name, email, password));
    }
  };

  return (
    <FormContainer>
      <h1>Sign Up</h1>
      {message && toast(`${message}`, { type: "error" })}
      {error && toast(`${error}`, { type: "error" })}
      {loading && <Loader />}
      <Form onSubmit={submitHandler}>
        <Form.Group controlId='name'>
          <Form.Label>Name</Form.Label>
          <Form.Control
            type='name'
            placeholder='Enter name'
            value={name}
            onChange={(e) => setName(e.target.value)}
          ></Form.Control>
        </Form.Group>
        <Form.Group controlId='email'>
          <Form.Label>Email Address</Form.Label>
          <Form.Control
            type='email'
            placeholder='Enter email'
            value={email}
            onChange={(e) => setEmail(e.target.value)}
          ></Form.Control>
        </Form.Group>

        <Form.Group controlId='password'>
          <Form.Label>Password</Form.Label>
          <Form.Control
            type='password'
            placeholder='Enter password'
            value={password}
            onChange={(e) => setPassword(e.target.value)}
          ></Form.Control>
        </Form.Group>

        <Form.Group controlId='confirmPassword'>
          <Form.Label>Confirm Password</Form.Label>
          <Form.Control
            type='confirmPassword'
            placeholder='Confirm password'
            value={confirmPassword}
            onChange={(e) => setConfirmPassword(e.target.value)}
          ></Form.Control>
        </Form.Group>

        <Button type='submit' variant='primary' className='my-2'>
          Register
        </Button>
      </Form>

      <Row className='py-3'>
        <Col>
          Have an account?{" "}
          <Link to={redirect ? `/login?redirect=${redirect}` : "/login"}>
            Login
          </Link>
        </Col>
      </Row>
    </FormContainer>
  );
};

export default RegisterScreen;
