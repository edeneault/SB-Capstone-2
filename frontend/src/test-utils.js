import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

// create a customRender that wraps the UI in a memory Router

const MemoryRouterWithInitialRoutes = ({ children, initialRoutes }) => {
  return <MemoryRouter initialEntries={initialRoutes}>{children}</MemoryRouter>;
};

const customRender = (ui, options) => {
  const initialRoutes =
    options && options.initialRoutes ? options.initialRoutes : ["/"];

  return render(ui, {
    wrapper: (args) =>
      MemoryRouterWithInitialRoutes({
        ...args,
        initialRoutes,
      }),
    ...options,
  });
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
