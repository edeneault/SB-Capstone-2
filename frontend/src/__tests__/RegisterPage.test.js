import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { initialState, render } from "../test-utils";
import { createMemoryHistory, createLocation } from "history";
import RegisterPage from "../pages/RegisterPage.js";

const mockStore = configureMockStore([thunk]);

describe("test", () => {
  const history = createMemoryHistory();
  const path = `/register`;

  const match = {
    isExact: false,
    path,
    url: path,
  };
  const location = createLocation(match.url);

  test("renders RegisterPage without crashing", () => {
    const store = mockStore(initialState);

    const { getByText } = render(
      <Provider store={store}>
        <RegisterPage location={location} history={history} />
      </Provider>,
      { initialRoutes: [`/register`] },
    );

    expect(getByText(/sign up/i)).toBeInTheDocument();
  });

  test("matches snapshot", function () {
    const store = mockStore(initialState);

    const { asFragment } = render(
      <Provider store={store}>
        <RegisterPage location={location} history={history} />
      </Provider>,
      { initialRoutes: [`/register`] },
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
