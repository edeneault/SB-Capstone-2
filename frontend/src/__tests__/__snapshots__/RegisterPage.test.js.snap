// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="justify-content-md-center row"
    >
      <div
        class="col-md-6 col-12"
      >
        <h1>
          Sign Up
        </h1>
        <form
          class=""
        >
          <div
            class="form-group"
          >
            <label
              class="form-label"
              for="name"
            >
              Name
            </label>
            <input
              class="form-control"
              id="name"
              placeholder="Enter name"
              type="name"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label
              class="form-label"
              for="email"
            >
              Email Address
            </label>
            <input
              class="form-control"
              id="email"
              placeholder="Enter email"
              type="email"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label
              class="form-label"
              for="password"
            >
              Password
            </label>
            <input
              class="form-control"
              id="password"
              placeholder="Enter password"
              type="password"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label
              class="form-label"
              for="confirmPassword"
            >
              Confirm Password
            </label>
            <input
              class="form-control"
              id="confirmPassword"
              placeholder="Confirm password"
              type="confirmPassword"
              value=""
            />
          </div>
          <button
            class="my-2 btn btn-primary"
            type="submit"
          >
            Register
          </button>
        </form>
        <div
          class="py-3 row"
        >
          <div
            class="col"
          >
            Have an account? 
            <a
              href="/login?redirect=/"
            >
              Login
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
